pkg/provider/aws/resources/eks_test.go

```diff
@@ -139,3 +139,91 @@ func TestNodeGroupNameFromConfig(t *testing.T) {
 		})
 	}
 }
+
+func Test_createNodeGroups(t *testing.T) {
+	cluster := &EksCluster{Name: "cluster"}
+	subnets := []*Subnet{
+		{Name: "private 1", Type: PrivateSubnet},
+		{Name: "private 2", Type: PrivateSubnet},
+		{Name: "public 1", Type: PublicSubnet},
+		{Name: "public 2", Type: PublicSubnet},
+	}
+	type NodeGroupExpect struct {
+		Name     string
+		DiskSize int
+		AmiType  string
+	}
+	tests := []struct {
+		name  string
+		units map[string]*config.ExecutionUnit
+		want  []NodeGroupExpect
+	}{
+		{
+			name: "no groups default",
+			units: map[string]*config.ExecutionUnit{
+				"a": {},
+				"b": {},
+			},
+			want: []NodeGroupExpect{
+				{Name: "private_t3_medium", DiskSize: 20, AmiType: "AL2_x86_64"},
+			},
+		},
+		{
+			name: "network types",
+			units: map[string]*config.ExecutionUnit{
+				"a": {},
+				"b": {NetworkPlacement: "public"},
+			},
+			want: []NodeGroupExpect{
+				{Name: "private_t3_medium", DiskSize: 20, AmiType: "AL2_x86_64"},
+				{Name: "public_t3_medium", DiskSize: 20, AmiType: "AL2_x86_64"},
+			},
+		},
+		{
+			name: "instance types",
+			units: map[string]*config.ExecutionUnit{
+				"a": {InfraParams: config.InfraParams{"instance_type": "c1.medium"}},
+				"b": {InfraParams: config.InfraParams{"instance_type": "g2.medium"}},
+			},
+			want: []NodeGroupExpect{
+				{Name: "private_c1_medium", DiskSize: 20, AmiType: "AL2_x86_64"},
+				{Name: "private_g2_medium", DiskSize: 20, AmiType: "AL2_x86_64_GPU"},
+			},
+		},
+		{
+			name: "disk size maximum",
+			units: map[string]*config.ExecutionUnit{
+				"a": {InfraParams: config.InfraParams{"disk_size_gib": 10}},
```

Comment:
try switching this to 100 and see if it has an effect, we want to make sure if its larger than the default its obeyed

Reply:
Unit `b` (the next line from this one) has the disk size of 50 to cover this